diff --git a/include/cute/stride.hpp b/include/cute/stride.hpp
--- a/include/cute/stride.hpp
+++ b/include/cute/stride.hpp
@@ -286,6 +286,19 @@ namespace detail {
 template <class Major>
 struct CompactLambda;
 
+template <class Major, class Shape>
+struct CompactSeq;
+
+template <class Shape>
+struct CompactSeq<LayoutLeft, Shape> {
+    using type = tuple_seq<Shape>;
+};
+
+template <class Shape>
+struct CompactSeq<LayoutRight, Shape> {
+    using type = tuple_rseq<Shape>;
+};
+
 // @pre is_integral<Current>
 // Return (result, current * product(shape)) to enable recurrence
 template <class Major, class Shape, class Current>
@@ -296,7 +309,7 @@ compact(Shape   const& shape,
 {
   if constexpr (is_tuple<Shape>::value) { // Shape::tuple Current::int
     using Lambda = CompactLambda<Major>;                  // Append or Prepend
-    using Seq    = typename Lambda::template seq<Shape>;  // Seq or RSeq
+    using Seq    = typename CompactSeq<Major, Shape>::type;  // Seq or RSeq
     return cute::detail::fold(shape, cute::make_tuple(cute::make_tuple(), current), Lambda{}, Seq{});
   } else {                                // Shape::int Current::int
     if constexpr (is_constant<1, Shape>::value) {
@@ -319,9 +332,6 @@ struct CompactLambda<LayoutLeft>
     auto result = detail::compact<LayoutLeft>(si, get<1>(init));
     return cute::make_tuple(append(get<0>(init), get<0>(result)), get<1>(result));  // Append
   }
-
-  template <class Shape>
-  using seq = tuple_seq<Shape>;                                                     // Seq
 };
 
 // For GCC8.5 -- Specialization LayoutRight
@@ -334,9 +344,6 @@ struct CompactLambda<LayoutRight>
     auto result = detail::compact<LayoutRight>(si, get<1>(init));
     return cute::make_tuple(prepend(get<0>(init), get<0>(result)), get<1>(result));  // Prepend
   }
-
-  template <class Shape>
-  using seq = tuple_rseq<Shape>;                                                     // RSeq
 };
 
 } // end namespace detail
